packages/fury-adapter-oas3-parser/test/unit/parser/oas/parseSchemaObject-test.js

```diff
@@ -533,4 +533,202 @@ describe('Schema Object', () => {
+      const schema = new namespace.elements.Object({
+        example: 'an example',
+      });
+      const parseResult = parse(context, schema);
+
+      expect(parseResult.length).to.equal(1);
+      expect(parseResult.get(0)).to.be.instanceof(namespace.elements.DataStructure);
+
+      const element = parseResult.get(0).content;
+      expect(element.attributes.get('samples').toValue()).to.deep.equal(['an example']);
+    });
+
+    it('adds an example value to the returned element when matching type', () => {
+      const schema = new namespace.elements.Object({
+        type: 'string',
+        example: 'an example',
+      });
+      const parseResult = parse(context, schema);
+
+      expect(parseResult.length).to.equal(1);
+      expect(parseResult.get(0)).to.be.instanceof(namespace.elements.DataStructure);
+
+      const element = parseResult.get(0).content;
+      expect(element.attributes.get('samples').toValue()).to.deep.equal(['an example']);
+    });
+
+    it('adds an example value to the returned element when matching enum', () => {
+      const schema = new namespace.elements.Object({
+        enum: ['an example'],
+        example: 'an example',
+      });
+      const parseResult = parse(context, schema);
+
+      expect(parseResult.length).to.equal(1);
+      expect(parseResult.get(0)).to.be.instanceof(namespace.elements.DataStructure);
+
+      const element = parseResult.get(0).content;
+      expect(element.attributes.get('samples').toValue()).to.deep.equal(['an example']);
+    });
+
+    it('allows a null example value when nullable is enabled', () => {
+      const schema = new namespace.elements.Object({
+        enum: ['n', 's', 'e', 'w'],
+        nullable: true,
+        example: null,
+      });
+      const parseResult = parse(context, schema);
+
+      expect(parseResult.length).to.equal(1);
+      expect(parseResult.get(0)).to.be.instanceof(namespace.elements.DataStructure);
+
+      const element = parseResult.get(0).content;
+      expect(element.attributes.get('samples').toValue()).to.deep.equal([null]);
+    });
+
+    it('warns when example does not match expected type', () => {
+      const schema = new namespace.elements.Object({
+        type: 'string',
+        example: true,
+      });
+      const parseResult = parse(context, schema);
+
+      expect(parseResult.length).to.equal(2);
+      expect(parseResult).to.contain.warning(
+        "'Schema Object' 'example' does not match expected type 'string'"
+      );
+    });
+
+    it('warns when example does not match enumeration value', () => {
+      const schema = new namespace.elements.Object({
+        type: 'number',
+        enum: [1, 2, 3],
+        example: 'north',
+      });
+      const parseResult = parse(context, schema);
+
+      expect(parseResult.length).to.equal(2);
+      expect(parseResult).to.contain.warning(
+        "'Schema Object' 'example' is not included in 'enum'"
+      );
+    });
+  });
```

Comment:
If having both default and example, will it validate both?

Reply:
Yes, but I can add a test.